[시저] Week2/Mid

caesar1030 opened the pull request:
추후 수정 예정

caesar1030 committed:
Refactor: 수정 컴포넌트 트리 구조

caesar1030 committed:
Feat: 추가 리스트 보기 좌우 화살표 기능

caesar1030 committed:
Feat: 추가 리스트 보기 기능

전체 언론사 및 내가 구독한 언론사 보기 기능 구현.

caesar1030 committed:
Docs: 수정 readme.md 파일

JaydenLee1116 commented on src/App.js:
```diff
@@ -1,36 +1,16 @@
+      <div class="news-stand"></div>
+      <div class="latest-news"></div>
+      <div class="contents"></div>
+    </div>`;
```
App class가 훨씬 깔끔해졌네요!
여기서 div 태그들보다 시멘틱 테그를 써보는 것도 좋을 거 같아요!

JaydenLee1116 commented on src/components/Contents.js:
```diff
@@ -0,0 +1,124 @@
+const ALL = "all";
+const SUBSCRIBING = "sub";
+const GRID = "grid";
+const LIST = "list";
```
위의 상수들을 따로 모아서 빼줘도 깔끔할 것 같습니다!

JaydenLee1116 commented on src/components/Contents.js:
```diff
@@ -0,0 +1,124 @@
+    if (!subscribingPresses) subscribingPresses = [];
+
+    try {
+      const res = await fetch("http://localhost:3001/presses");
```
url도 따로 변수에 지정해두면 좋을 것 같아요!

JaydenLee1116 commented on src/components/Contents.js:
```diff
@@ -0,0 +1,124 @@
+      subscribingPresses: [],
+      presses: [],
+      viewOption: LIST,
+      subscriptionOption: ALL,
```
옹... 로직은 더 봐야하겠지만, viewoption과 subscriptionOption을 나눠서 상태를 관리하는 게 좋아보여요! 경우의 수가 2*2개 될텐데, 저도 이 부분을 고민해봐야겠어요!

JaydenLee1116 commented on src/components/GridView.js:
```diff
@@ -0,0 +1,104 @@
+const INITIAL_PAGE_NUM = 0;
+
+const LEFT = -1;
+const RIGHT = 1;
```
여기도 상수 빼주기!

JaydenLee1116 commented on src/components/Logo.js:
```diff
@@ -34,19 +20,37 @@ export default class Cell extends Component {
+    const name = this.props.name;
+    const src = this.props.src;
+    return `
+    <img class="news-list__image" ${src ? `src=${src}` : ""} ${
```
```js
return `<img class="news-list__image" src="${src ? src : ""}"}/>`
```
저는 이게 좀더 깔끔한 느낌인데, 시저는 어떻게 생각하실지 궁금하네요!

JaydenLee1116 commented on src/components/NewsContent.js:
```diff
@@ -0,0 +1,76 @@
+                <ul class="sub-news">
+                  ${listHtml}
+                </ul>
+                <span class="caption">UPI뉴스 언론사에서 직접 편집한 뉴스입니다.</span>
```
이 문장도 fetch한 데이터의 press(언론사)에 따라 변하지 않나요..?!

JaydenLee1116 reviewed (commented):
시저~~~ 모든 코드를 다 본건 아니지만, 정말 리액트랑 비슷한 느낌인 것 같아요!
넘넘 수고하셨습니다~!

crongro reviewed (changes requested):
componentDidMount, componentDidUpdate 는 리액트에서 deprecated 된 메서드라 좀 구식입니다.(참고)
하지만 이런 시도는 좋네요.

/utils/storageManger.js 로 디렉토리와 네이밍을 바꾸는것도 좋을듯. 다른 유틸도 저 아래 만들면 되니까.

components 디렉토리가 한 단계 계층으로 나열되어 있는데, 만약 실제로 이것이 계층적으로 (위-아래 구조)되어있다면 디렉토리를 구성해서 하위개념으로 구조를 만들어도  될 듯. 

renderChildComponents 와 같은 함수의 복잡도를 어떻게 낮출지 고민.

이벤트 핸들러는 가급적 메서드로 분리. 클래스에 그런 핸들러 메서드가 있어도 괜찮습니다.


컴포넌트 아래 뷰 들의 모습이 일관된 형태라 좋네요.
이벤트 핸들러 분리, 복잡한 함수 메서드 나누기, 디렉토리 구조는 수정이라기보다 검토 해보세요. (단계별 디렉토리 구성)
이부분 확인하고 넘어갑시다.

caesar1030 committed:
Refactor: 수정 utils / 폴더 구조 / 이벤트 핸들러 분리

storageManage의 경우 util 폴더로 분리.
폴더 구조의 경우 필요성을 느껴 시도 했지만 버튼의 경우 그리드 뷰와
리스트 뷰에서 동시에 자식 컴포넌트로 들어가는 문제 때문에 마음에 들지
않음.
이벤트 핸들러 변수로 따로 분리. 가독성 및 테스트가 이유.

crongro reviewed (approved):
자잘한 수정 잘 확인했어요. 
머지고고
